crates/astria-sequencer/src/accounts/ics20_transfer.rs

```diff
@@ -350,16 +371,23 @@ async fn execute_ics20_transfer<S: StateWriteExt>(
             format!("{dest_port}/{dest_channel}/{}", packet_data.denom)
         };
 
-        // TODO(https://github.com/astriaorg/astria/issues/603): register denomination
-        // in global ID -> denom map if it's not already there
+        let asset: IbcAsset = prefixed_denomination
+            .parse()
+            .context("failed to parse prefixed denomination as IbcAsset")?;
+
+        // register denomination in global ID -> denom map if it's not already there
+        if state.get_ibc_asset(asset.id()).await.is_err() {
+            state
+                .put_ibc_asset(asset.clone())
+                .context("failed to put IBC asset in storage")?;
+        }
 
-        let asset_id = Id::from_denom(&prefixed_denomination);
         let user_balance = state
-            .get_account_balance(recipient, asset_id)
+            .get_account_balance(recipient, asset.id())
             .await
             .context("failed to get user account balance in execute_ics20_transfer")?;
         state
-            .put_account_balance(recipient, asset_id, user_balance + packet_amount)
+            .put_account_balance(recipient, asset.id(), user_balance + packet_amount)
```

Comment:
as for number 2, no because this is executed within its own `begin_transaction` scope on the `StateWrite` which will not be committed if anything inside the transaction execution fails